tools/buf/buf.yaml

```diff
@@ -27,6 +27,11 @@ lint:
       - plugins/shared/structs/proto/attribute.proto
       - plugins/shared/structs/proto/recoverable_error.proto
       - plugins/shared/structs/proto/stats.proto
+    FIELD_LOWER_SNAKE_CASE:
+      - plugins/base/proto/base.proto
+      - plugins/device/proto/device.proto
+      - plugins/drivers/proto/driver.proto
+      - plugins/shared/hclspec/hcl_spec.proto
```

Comment:
I feel mixed here about these ignores.  Ideally, we only ignore the current appearances, as we are bound by compatibility, but we shouldn't be adding new fields that violate this check in these files.

I considered using targetted fine-grain ignore comments with [`allow_comments_ignore`](https://docs.buf.build/lint-configuration#allow_comment_ignores) to achieve this, but the flag is discouraged.  I would love to some feedback/input before switching to that.

Reply:
I agree; we should avoid adding new entries here.

Reply:
I took a quick pass thru and it looks like there's very few existing violations. It seems like it might be worth it to do the fine-grain ignore (and mark it as being for backwards compatibility purposes).

Reply:
Sounds good. I'll convert these and the rest of the ignores in this config file into comments as it seems fit then.